### PR TITLE
method overloading fix for enum argument

### PR DESCRIPTION
**Pull request by denfromufa**

### CODE REVIEW

File: src/runtime/methodbinder.cs [hunk 1]
```diff
@@ -309,26 +309,38 @@ internal static MethodInfo MatchSignatureAndParameters(MethodInfo[] mi, Type[] g
...
                                         }
                                     }
-                                    else {
-                                        Runtime.Decref(pyoptype);
+                                    if (!typematch) {
+                                        // this takes care of enum values
+                                        TypeCode argtypecode = Type.GetTypeCode(pi[n].ParameterType);
+                                        Type typearg = Type.GetType("System." + argtypecode.ToString());
+                                        if (typearg == clrtype) {
+                                            typematch = true;
```
Comment (tonyroberts) [inline src/runtime/methodbinder.cs]: Would comparing the type code of the parameter type and the clrtype be more appropriate here?

Comment (denfromufa) [inline src/runtime/methodbinder.cs]: @tonyroberts done! thanks for pointing out cleaner solution. I also added this "regression" test.

**Comment by tonyroberts** [comment]
@denfromufa great, thanks.

**Comment by stonebig** [comment]
a "dev2" release this wek-end ?

**Pull request merged**